### PR TITLE
[libc][docs] split "String Functions" into string.h, strings.h, stdlib.h, and inttypes.h

### PR DESCRIPTION
**Pull request by nickdesaulniers**
bcmp, bcopy, and bzero should be moved from libc/src/string/ to
lib/src/strings/ in order for docgen to use existing conventions to find
whether we implement a function or not.

We should add support to docgen for mentioning glibc extensions (mempcpy) or
extensions from other libcs.

### CODE REVIEW

**Comment by nickdesaulniers** [comment]
Note to reviewers, this PR is stacked on top of https://github.com/llvm/llvm-project/pull/118836 AND https://github.com/llvm/llvm-project/pull/118852. The first 2 commits should be skipped in review for this PR.

**Review by michaelrj-google — COMMENTED** [review]
I'd like to wait until the `strings.h` situation is handled before landing this.

**Comment by nickdesaulniers** [comment]
SGTM; filed https://github.com/llvm/llvm-project/issues/118875. Cleaning that up can land before this then.

**Comment by nickdesaulniers** [comment]
https://github.com/llvm/llvm-project/pull/118899

File: libc/utils/docgen/strings.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    "bzero": {
+      "posix-definition": "removed in POSIX.1-2008"
+    },
+    "bcmp": {
+      "posix-definition": "removed in POSIX.1-2008"
+    },
+    "bcopy": {
+      "posix-definition": "removed in POSIX.1-2008"
+    }
+  }
```
Comment (nickdesaulniers) [inline libc/utils/docgen/strings.json:12]: need to add `index`, `rindex`, `strcasecmp`, and `strncasecmp`.

**Review by michaelrj-google — APPROVED** [review]
LGTM now that the `strings` split has landed

**Pull request merged**